identity-server/src/IdentityServer/Configuration/DependencyInjection/Options/DynamicProviderOptions.cs

```diff
@@ -22,6 +22,12 @@ public class DynamicProviderOptions
     /// </summary>
     public PathString PathPrefix { get; set; } = "/federation";
 
+    /// <summary>
+    /// If set, this callback will be used over <see cref="PathPrefix"/> to determine if the request is for a dynamic provider.
+    /// If the request is a match, the callback should return the name of the scheme which matched.
+    /// </summary>
+    public Func<HttpContext, string?>? PathMatchingCallback { get; set; }
```

Comment:
Should this be async? Do we think users would want to interact with a data store in the callback?